app/controllers/posts_controller.rb

```diff
@@ -243,16 +243,22 @@ def change_status
       flash[:error] = "Invalid status selected."
     else
       @post.status = new_status
-      @post.save
-      flash[:success] = "Post has been marked #{params[:status]}."
+      if @post.save
+        flash[:success] = "Post has been marked #{params[:status]}."
+      else
+        flash[:error] = "Status could not be updated."
```

Comment:
Maybe show the post's full list of errors too? It might give a better idea of what the exact error is.